fchat/characters.ts

```diff
@@ -273,16 +320,52 @@ export default function (this: void, connection: Connection): Interfaces.State {
           state.bookmarks.splice(state.bookmarks.indexOf(character), 1);
         break;
       case 'friendadd':
-        if (character.isFriend) return;
+        // Always add to global friends
         state.friendList.push(data.name);
         character.isFriend = true;
         if (character.status !== 'offline') state.friends.push(character);
+
+        // If per-character friends is enabled, also update that list
```

Comment:
Isn't this just part of ``refreshFriends()`` again? Just make it a helper function.